sklearn/metrics/ranking.py

```diff
@@ -410,7 +413,20 @@ def precision_recall_curve(y_true, probas_pred, pos_label=None,
     # and reverse the outputs so recall is decreasing
     last_ind = tps.searchsorted(tps[-1])
     sl = slice(last_ind, None, -1)
-    return np.r_[precision[sl], 1], np.r_[recall[sl], 0], thresholds[sl]
+
+    if interpolate:
+        prec = np.r_[precision[sl], 1]
+        p_temp = prec[0]
+        n = len(prec)
+        for i in range(n):
+            if prec[i] < p_temp:
+                prec[i] = p_temp
+            else:
+                p_temp = prec[i]
+        return prec, np.r_[recall[sl], 0], thresholds[sl]
```

Comment:
sure. This is ok. I am just asking why not using:

http://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html#scipy.interpolate.interp1d

?